src/test/java/tr/com/orioninc/laborant/service/AdminServiceTest.java

```diff
@@ -0,0 +1,175 @@
+package tr.com.orioninc.laborant.service;
+
+import lombok.extern.log4j.Log4j2;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import tr.com.orioninc.laborant.model.Lab;
+import tr.com.orioninc.laborant.repository.LabRepository;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+@Log4j2
+@DataJpaTest
+class AdminServiceTest {
+
+
+    @Autowired
+    private LabRepository labRepository;
+    private AdminService underTest;
+
+
+    @BeforeEach
+    void setUp() {
+        underTest = new AdminService(labRepository);
+    }
+
+    @Test
+    void canGetAllLabs() {
+        // when
+        List<Lab> test = underTest.getAllLabs();
+        // then
+        assertEquals(labRepository.findAll(), test);
+    }
+
+    @Test
+    void canFindLabByName() {
+        // given
+        String labName = "testLab";
+        Lab lab = new Lab(
+                labName, "testUser", "testPassword", "testHost", 22);
+        labRepository.save(lab);
+        // when
+        Lab foundLab = underTest.findLabByName(labName);
+        // then
+        assertEquals(lab, foundLab);
+    }
+
+    @Test
+    void canUpdateLabByName() {
+        // given
+        String labName = "testLab";
+        Lab lab = new Lab(
+                labName, "testUser", "testPassword", "testHost", 22);
+        labRepository.save(lab);
+        log.info("Lab saved: {}", lab);
+        Lab toBeUpdated = new Lab(
+                labName, "update", "update", "update", 222);
+        // when
+        underTest.updateLabByName(labName, toBeUpdated);
+        // then
+        assertEquals(underTest.getLab(labName).getLabName(), toBeUpdated.getLabName());
+        assertEquals(underTest.getLab(labName).getUserName(), toBeUpdated.getUserName());
+        assertEquals(underTest.getLab(labName).getPassword(), toBeUpdated.getPassword());
+        assertEquals(underTest.getLab(labName).getHost(), toBeUpdated.getHost());
+        assertEquals(underTest.getLab(labName).getPort(), toBeUpdated.getPort());
```

Comment:
You can refactor this with:
assertThat(foundLab).isNotNull()
.returns("testUser, Lab::getUser)
.returns(.....